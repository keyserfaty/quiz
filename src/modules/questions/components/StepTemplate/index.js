import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'

import styles from './statics/styles'

const StepTemplate = props => {
  const { style, children, header, footer } = props

  return (
    <View style={[styles.container, style]}>
      <View style={styles.header}>
        {header}
      </View>
      <View style={styles.body}>
        {children}
      </View>
      <View style={styles.footer}>
        {footer}
      </View>
    </View>
  )
}

StepTemplate.propTypes = {
  style: PropTypes.array,
  header: PropTypes.any,
  footer: PropTypes.any,
  children: PropTypes.any
}

export default StepTemplate