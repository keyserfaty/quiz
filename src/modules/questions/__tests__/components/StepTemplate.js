import React from 'react'
import { View, Text } from 'react-native'

import StepTemplate from '../../components/StepTemplate'

import Shallow from 'react-test-renderer/shallow'

const renderer = new Shallow()

test('Should render', () => {
  const component = renderer.render(<StepTemplate />)
  expect(component).toMatchSnapshot()
})

test('Should render contents', () => {
  const component = renderer.render(<StepTemplate header='header' footer='footer' />)
  expect(component).toMatchSnapshot()
})

test('Should render children', () => {
  const component = renderer.render(
    <StepTemplate header='header' footer='footer'>
      <View>
        <Text>Some child</Text>
      </View>
    </StepTemplate>
  )

  expect(component).toMatchSnapshot()
})
