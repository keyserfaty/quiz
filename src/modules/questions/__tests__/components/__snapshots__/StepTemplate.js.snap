// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "space-between",
        "padding": 60,
      },
      undefined,
    ]
  }
>
  <Component
    style={Object {}}
  />
  <Component
    style={Object {}}
  />
  <Component
    style={Object {}}
  />
</Component>
`;

exports[`Should render children 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "space-between",
        "padding": 60,
      },
      undefined,
    ]
  }
>
  <Component
    style={Object {}}
  >
    header
  </Component>
  <Component
    style={Object {}}
  >
    <Component>
      <Component>
        Some child
      </Component>
    </Component>
  </Component>
  <Component
    style={Object {}}
  >
    footer
  </Component>
</Component>
`;

exports[`Should render contents 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "space-between",
        "padding": 60,
      },
      undefined,
    ]
  }
>
  <Component
    style={Object {}}
  >
    header
  </Component>
  <Component
    style={Object {}}
  />
  <Component
    style={Object {}}
  >
    footer
  </Component>
</Component>
`;
